nav2_costmap_2d/plugins/static_layer.cpp

```diff
@@ -124,10 +124,18 @@ StaticLayer::getParameters()
   declareParameter("subscribe_to_updates", rclcpp::ParameterValue(false));
   declareParameter("map_subscribe_transient_local", rclcpp::ParameterValue(true));
   declareParameter("transform_tolerance", rclcpp::ParameterValue(0.0));
+  declareParameter("map_topic", rclcpp::ParameterValue(""));
 
   node_->get_parameter(name_ + "." + "enabled", enabled_);
   node_->get_parameter(name_ + "." + "subscribe_to_updates", subscribe_to_updates_);
-  node_->get_parameter("map_topic", map_topic_);
+  std::string private_map_topic, global_map_topic;
+  node_->get_parameter(name_ + "." + "map_topic", private_map_topic);
```

Comment:
Why not make them all private and remove the global one? I don't have a particular issue with this but curious the justification for it.

Reply:
I thought it would be good to keep it compatible with the global parameter so the user can choose which one to use depending on whatever is the most convenient. In the future if any layers are added that also subscribe to map topics like the static layer then the global parameter may be advantageous whereas the private parameter comes in handy if you need to have subscriptions to an ancillary map topics.